stock_location_children/models/stock_location.py

```diff
@@ -0,0 +1,44 @@
+# Copyright 2020 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
+from odoo import api, fields, models
+
+
+class StockLocation(models.Model):
+
+    _inherit = "stock.location"
+
+    children_ids = fields.Many2many(
+        "stock.location",
+        "stock_location_children_ids",
+        "parent_id",
+        "children_id",
+        compute="_compute_children_ids",
+        store=True,
+        help="All the children (multi-level) stock location of this location",
+    )
+
+    @api.depends("child_ids", "child_ids.children_ids")
```

Comment:
shouldn't be `child_ids.child_ids`? :thinking:

Reply:
or better `parent_path`

Reply:
Test code:

```
>>> Loc = self.env['stock.location']
>>> stock = Loc.browse(8)
>>> t1 = Loc.create({'name': 'test1', 'location_id': stock.id})
>>> t2 = Loc.create({'name': 'test2', 'location_id': t1.id})
>>> t3 = Loc.create({'name': 'test2', 'location_id': t2.id})
>>> self.env.cr.commit()
```
With api.depends('child_ids', 'child_ids.child_ids'), the computation method is called once with self = stock.location(7, 8, 26, 27, 28)

With api.depends('child_ids', 'child_ids.children_ids', the computation method is called 6 times with self containing a single location each time. 

@grindtildeath  My advice: prefer child_ids.child_ids if it does not break any known test case as it seems more efficient. 

@simahawk I'm not sure how to depend on parent_path. Can you suggest something?

Reply:
@gurneyalex thanks for testing, I suspected this behavior.
Regarding `parent_path`: never mind. I did a quick check and I don't think is going to work because it's handled in "magic" methods.